src/transformers/models/superglue/modeling_superglue.py

```diff
@@ -0,0 +1,845 @@
+    keypoints is not fixed and can vary from image to image, which makes batching non-trivial. In the batch of images,
+    the maximum number of matches is set as the dimension of the matches and matching scores. The mask
+    tensor is used to indicate which values in the keypoints, matches and matching_scores tensors are keypoint matching
+    information.
+
+    Args:
+        loss (`torch.FloatTensor` of shape `(1,)`, *optional*):
+            Loss computed during training.
+        mask (`torch.BoolTensor` of shape `(batch_size, num_keypoints)`):
+            Mask indicating which values in matches and matching_scores are keypoint matching information.
+        matches (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Index of keypoint matched in the other image.
+        matching_scores (`torch.FloatTensor` of shape `(batch_size, 2, num_matches)`):
+            Scores of predicted matches.
+        keypoints (`torch.FloatTensor` of shape `(batch_size, num_keypoints, 2)`):
+            Absolute (x, y) coordinates of predicted keypoints in a given image.
+    """
+
+    loss: Optional[torch.FloatTensor] = None
+    mask: torch.FloatTensor = None
+    matches: torch.FloatTensor = None
+    matching_scores: torch.FloatTensor = None
+    keypoints: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+class SuperGlueMultiLayerPerceptron(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+        channels: List[int],
+        do_batch_norm: bool = True,
+    ):
+        super().__init__()
+        num_layers = len(channels)
+        layers = []
+        for i in range(1, num_layers):
+            layers.append(nn.Conv1d(channels[i - 1], channels[i], kernel_size=1, bias=True))
+            if i < (num_layers - 1):
+                if do_batch_norm:
+                    layers.append(nn.BatchNorm1d(channels[i]))
+                layers.append(nn.ReLU())
+        self.layers = nn.Sequential(*layers)
+        nn.init.constant_(self.layers[-1].bias, 0.0)
+
+    def forward(
+        self,
+        input: torch.Tensor,
+        output_hidden_states: Optional[bool] = False,
+    ) -> Union[Tuple, torch.Tensor]:
+        all_hidden_states = () if output_hidden_states else None
+        for layer in self.layers:
+            input = layer(input)
+            if output_hidden_states and isinstance(layer, nn.Conv1d):
+                all_hidden_states = all_hidden_states + (input,)
+        output = input
+        return output, all_hidden_states
+
+
+class SuperGlueKeypointEncoder(nn.Module):
+    def __init__(
+        self,
+        config: SuperGlueConfig,
+    ):
+        super().__init__()
+        self.layer_sizes = config.keypoint_encoder_sizes
+        self.feature_dim = config.descriptor_dim
+        self.encoder = SuperGlueMultiLayerPerceptron(config, [3] + self.layer_sizes + [self.feature_dim])
```

Comment:
no need to set these as attrs 

```suggestion
        layer_sizes = config.keypoint_encoder_sizes
        feature_dim = config.descriptor_dim
        self.encoder = SuperGlueMultiLayerPerceptron(config, [3] + layer_sizes + [feature_dim])
```

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/29886/commits/601da1cf0756cef7339d4dd33faeefbfcd073ac9)